### DISCUSSION TITLE
Can I duplicate a setting?

### DISCUSSION THREAD

**Discussion by feedhopper** [chhoumann/quickadd#309] [Q&A] [answered]
I would like to create identical or marginally differing setups. Is there a way to duplicate a setup?

This is not a feature request per se as I am not sure this is a need that justifies the work that would go into programming a dedicated button.

Rather, I wondered if there is any way to manually copy the settings for one setup and merely altering the name by hand. Would this work? And: Which settings would I need to manipulate? Where are they stored?

**Comment by generic-user** [comment, marked as answer]
I think they are stored here: `vault\.obsidian\plugins\quickadd\data.json`.

```json
    {
      "id": "2002b13d-9601-4e59-957b-2731663e17e0",
      "name": "RS.COOKING.SOUP",
      "type": "Template",
      "command": false,
      "templatePath": "assets/templates/t.Recept.md",
      "fileNameFormat": {
        "enabled": true,
        "format": "RS.COOKING.SOUP.{{NAME}}"
      },
```

**Comment by chhoumann** [comment]
Would recommend copying and pasting the JSON objects & then just changing the IDs for now, as @generic-user suggests.

Later, the plan is to create import/export/copy/paste functionality.

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

Both of the plans mentioned here shipped, so no data.json surgery is needed anymore: every choice has a one-click **Duplicate** (also in the right-click menu in Settings -> QuickAdd), and the copy gets fresh ids automatically. That landed in v0.10.0, a couple of months after this thread.

For moving setups between vaults or sharing them, there is also choice export/import via packages: https://quickadd.obsidian.guide/docs/Choices/Packages

Closing as resolved.
